api/src/main/java/io/minio/credentials/AssumeRoleProvider.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio.credentials;
+
+import io.minio.Digest;
+import io.minio.Signer;
+import io.minio.Time;
+import io.minio.Xml;
+import io.minio.errors.XmlParserException;
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.simpleframework.xml.Element;
+import org.simpleframework.xml.Namespace;
+import org.simpleframework.xml.Path;
+import org.simpleframework.xml.Root;
+
+/**
+ * Credential provider using <a
+ * href="https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html">AssumeRole
+ * API</a>.
+ */
+public class AssumeRoleProvider implements Provider {
+  public static final int DEFAULT_DURATION_SECONDS = (int) TimeUnit.HOURS.toSeconds(1);
+  private final HttpUrl stsEndpoint;
+  private final String accessKey;
+  private final String secretKey;
+  private final String region;
+  private final OkHttpClient httpClient;
+  private final String contentSha256;
+  private final Request request;
+  private Credentials credentials;
```

Comment:
The fields `stsEndpoint`, `httpClient` and `credentials` seem to be common with `WebIdentityClientGrantsProvider`. We can consider adding an abstract base class that has a constructor that takes these arguments and call it from the constructors of both `WebIdentityClientGrantsProvider` and `AssumeRoleProvider`